### PR TITLE
Fix MSVC warning in std::chrono::time_point formatter

### PR DESCRIPTION
**Pull request by hmbj**
The condition is constexpr causing MSVC level 4 warning:
    warning C4127: conditional expression is constant

Made condition constexpr to eliminate the warning.

Code to reproduce the issue:
```
#include <fmt/chrono.h>
#include <fmt/xchar.h>
int main()
{
    fmt::format(L"{}", std::chrono::system_clock::time_point());
}
```

CMakeLists.txt:
```
cmake_minimum_required(VERSION 3.14)
project(mytest LANGUAGES CXX)

include(FetchContent)
FetchContent_Declare(fmt
  GIT_REPOSITORY  https://github.com/fmtlib/fmt.git
  GIT_TAG         10.0.0
)
FetchContent_MakeAvailable(fmt)

add_compile_options("$<$<CXX_COMPILER_ID:MSVC>:/W4>")
add_executable(mytest main.cpp)
target_compile_features(mytest PRIVATE cxx_std_17)
target_link_libraries(mytest PRIVATE fmt::fmt-header-only)
```

### CODE REVIEW

File: include/fmt/chrono.h [hunk 1]
```diff
@@ -2153,10 +2154,10 @@ struct formatter<std::chrono::time_point<std::chrono::system_clock, Duration>,
       return formatter<std::tm, Char>::do_format(
           gmtime(std::chrono::time_point_cast<std::chrono::seconds>(val)), ctx,
           &subsecs);
+    } else {
+      return formatter<std::tm, Char>::format(
+          gmtime(std::chrono::time_point_cast<std::chrono::seconds>(val)), ctx);
     }
```
Comment (vitaut) [inline include/fmt/chrono.h:2160]: This looks unrelated, please revert.

Comment (hmbj) [reply include/fmt/chrono.h]: Used to get an unreachable code warning without the else. Pushed a revert to see what the pipeline checks says...

**Comment by vitaut** [comment]
Thanks
